src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -810,31 +811,62 @@ private LanguageExpression ConvertBinary(BinaryOperationSyntax syntax)
 
         private LanguageExpression ConvertUnary(UnaryOperationSyntax syntax)
         {
-            LanguageExpression convertedOperand = ConvertExpression(syntax.Expression);
-
             switch (syntax.Operator)
             {
                 case UnaryOperator.Not:
+                    LanguageExpression convertedOperand = ConvertExpression(syntax.Expression);
                     return CreateFunction("not", convertedOperand);
 
                 case UnaryOperator.Minus:
-                    if (convertedOperand is JTokenExpression literal && literal.Value.Type == JTokenType.Integer)
+                    if (syntax.Expression is IntegerLiteralSyntax integerLiteral)
                     {
-                        // invert the integer literal
-                        int literalValue = literal.Value.Value<int>();
-                        return new JTokenExpression(-literalValue);
+                        //do codegen logic
+                        LanguageExpression integerExpression = ConvertInteger(integerLiteral, true);
+                        return integerExpression;
                     }
 
                     return CreateFunction(
                         "sub",
                         new JTokenExpression(0),
-                        convertedOperand);
+                        ConvertExpression(syntax.Expression));
 
                 default:
                     throw new NotImplementedException($"Cannot emit unexpected unary operator '{syntax.Operator}.");
             }
         }
 
+        // for 32 bit integers, we return the integer value
```

Comment:
It would be good to explain the 'why' as well as the 'what' here for the reason to use the "json" function - e.g. "because the deployment engine can only handle 32-bit integers expressed as literal values"